lib/in_out/parser.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import re
+
+from parser_regex import log_regex
+
+
+class Parser:
+    def __init__(self, client):
+
+        # self.nicks maintains a list of current user nicks present \
+        # on channels. We will use this heuristic to see nicks mention \
+        # in a message. User mention can happen in following ways: \
+        # <rohit> bhuvan: or <rohit> bhuvan, kausam, or
+        # <rohit> nice work kausam.
+
+        self.nicks = []
+        self.regexes = log_regex[client]
```

Comment:
client is not part of the log_regex array put up in `parser_regex.py`. I guess you mean the client to be the name of the IRC channel. If so, it is better to call the variable as `channel`.

Reply:
done